src/main/java/seedu/address/model/pet/AttendanceHashMap.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.model.pet;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.util.Collection;
+import java.util.HashMap;
+
+import seedu.address.model.attendance.AttendanceEntry;
+
+/**
+ * Represents a Pet's attendance history in the address book.
+ * Guarantees: immutable; is always valid.
+ */
+public class AttendanceHashMap {
+    public final HashMap<LocalDate, AttendanceEntry> attendanceHashMap;
```

Comment:
Maybe can consider to make attendanceHashMap non-final so that methods to add/delete attendance can be done without creating a temporary HashMap. Or is there a reason you chose to make it final?

Reply:
Will remove usages of temporary HashMaps. Originally I made them `immutable` to allow for ease of debugging, but there is no need now. However, we might want to check with the tutor regarding this, as I realised that most of the fields in AB3 are immutable.

Reply:
True, thanks for the clarification! Yes we should check with the tutor.